File: AkitaModel/Robin/IReportDataObject.cs
namespace Skyware.Lis.AkitaModel.Robin;

/// <summary>
/// Data object that can be encoded as Base64.
/// </summary>
public interface IReportDataObject
{

    /// <summary>
    /// Serialize to JSON and converts it to Base64 string.
    /// </summary>
    /// <returns></returns>
    string GetBase64Data();

}

File: AkitaMotelTS/src/Robin/IReportDataObject.ts
/**
 * Skyware.Lis.AkitaModel.Robin
 */

/**
 * Data object that can be encoded as Base64.
 */
export interface IReportDataObject {
  /**
   * Serialize to JSON and converts it to Base64 string.
   */
  getBase64Data(): string;
}
